internal/controller/pkg/revision/dependency.go

```diff
@@ -67,7 +68,7 @@ func NewPackageDependencyManager(c client.Client, nd dag.NewDAGFn, t v1alpha1.Pa
 
 // Resolve resolves package dependencies.
 func (m *PackageDependencyManager) Resolve(ctx context.Context, pkg runtime.Object, pr v1.PackageRevision) (found, installed, invalid int, err error) { // nolint:gocyclo
-	pack, ok := pkg.(pkgmeta.Pkg)
+	pack, ok := xpkg.ConvertToPkg(pkg, &pkgmeta.Provider{}, &pkgmeta.Configuration{})
```

Comment:
Yeah, I started out with the call being there then realised we also needed conversion in the linter calls which happen before this. I've now updated it so that we make a call here and everywhere else we might need conversion, but we only convert once (redundant calls will skip conversion).